src/components/TableWithInlineEditing.js

```diff
@@ -0,0 +1,211 @@
+import React, { Fragment, useState } from 'react';
+import PropTypes from 'prop-types';
+import { TableCell, TableRow, Table, TableHead, TableBody } from '@material-ui/core';
+import { InputField, Dropdown } from '@linn-it/linn-form-components-library';
+import { makeStyles } from '@material-ui/styles';
+import AddIcon from '@material-ui/icons/Add'; //will be adding these two in soon
+import DeleteIcon from '@material-ui/icons/Delete';
+
+function TableWithInlineEditing({ content, columnsInfo, updateContent, allowedToEdit }) {
+    const [editingCellId, setEditingCellId] = useState({});
+
+    const handleRowChange = (propertyName, newValue, rowIndex) => {
+        const updatedRow = { ...content[rowIndex], [propertyName]: newValue };
+
+        updateContent(
+            content
+                .slice(0, rowIndex)
+                .concat(updatedRow)
+                .concat(content.slice(rowIndex + 1, content.length + 1))
+        );
+    };
+
+    const switchToEditingDifferentCell = newCellId => {
+        setEditingCellId(newCellId);
+    };
+
+    const clearEditingCell = () => {
+        setEditingCellId('');
+    };
+
+    return (
+        <Fragment>
+            <Table>
+                <TableHead key="headers" onClick={clearEditingCell}>
+                    <TableRow>
+                        {columnsInfo.map(el => (
+                            <TableCell>{el.title}</TableCell>
+                        ))}
+                    </TableRow>
+                </TableHead>
+                <TableBody>
+                    {content.map((el, index) => (
+                        <Row
+                            rowContent={el}
+                            key={index}
```

Comment:
react will probably complain if you use a map index as a key... Does el have some unique property that could be used as a key?

Reply:
yeah it does complain but unfortunately there's nothing we can use as the key I don't think - we don't know which properties we'll be passing into the table and which of them would be unique. Looks like we've done the same in other table components, although I think I'll have to come up with something when I add a delete option in, likely have to define which is unique each time we use the component. I had tried using an npm package I saw online that was meant to solve this but it didn't work either

Reply:
I think you should insist the data contains an id field. We do this for a few of the shared components. 
I think it's fair to have to possibly modify your data in return for the convenience. 

The biggest culprit for this is probably the reportTable at the moment. It's on my list to fix that.

Reply:
I've never really had great success with this either. You could just force parents to pass in data with well defined unique ids perhaps but might be a bit of a pain.

Reply:
Didn't see Richards comment before I typed a not as clear version of the same thing..

Reply:
yeah forcing parents to pass an id field in sounds good, I'll put that in